CMakeLists.txt

```diff
@@ -1,120 +1,121 @@
-cmake_minimum_required(VERSION 2.8.8)
-project(instr)
-
-find_package(LLVM REQUIRED CONFIG)
-if (${LLVM_PACKAGE_VERSION} VERSION_GREATER "3.4")
-  set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${LLVM_DIR}")
-  include(LLVMConfig)
+cmake_minimum_required(VERSION 3.1.0)
+project(sbt-instrumentation CXX)
+
+# --------------------------------------------------
+# Build type
+# --------------------------------------------------
+if(NOT CMAKE_BUILD_TYPE)
+  message(STATUS "Build type not set. Setting default.")
+  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "" FORCE)
 endif()
+message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
+set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "RelWithDebInfo"
+                                                     "MinSizeRel" "Release")
+
+# --------------------------------------------------
+# Compilation flags
+# --------------------------------------------------
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -fno-rtti")
+
+# --------------------------------------------------
+# LLVM
+# --------------------------------------------------
+find_package(LLVM 3.9.1 REQUIRED CONFIG)
```

Comment:
This change seems to break compilation:

> CMake Error at CMakeLists.txt:25 (find_package):
  Could not find a configuration file for package "LLVM" that is compatible
  with requested version "3.9.1".  
>
> The following configuration files were considered but not accepted:
>
>    /usr/lib/llvm-11/cmake/LLVMConfig.cmake, version: 11.1.0
    /usr/lib/llvm-9/cmake/LLVMConfig.cmake, version: 9.0.1
    /usr/lib/llvm-8/cmake/LLVMConfig.cmake, version: 8.0.1
    /usr/lib/llvm-7/cmake/LLVMConfig.cmake, version: 7.0.1
    /usr/lib/llvm-10/cmake/LLVMConfig.cmake, version: 10.0.1